Updates for a variety of diagnostic docs (#31001)
dotnet/docs: docs/core/diagnostics/metrics-instrumentation.md

  increments and decrements using <xref:System.Diagnostics.Metrics.UpDownCounter%601.Add%2A>. For example, assume that the caller invokes `Add()` once each second with successive
  values 1, 5, -2, 3, -1, -3. If the collection tool updates every three seconds, then the total after three seconds is 1+5-2=4 and the total after six seconds is 1+5-2+3-1-3=3.

- **ObservableCounter** (<xref:System.Diagnostics.Metrics.Meter.CreateObservableCounter%2A>) - This instrument is similar to Counter except that the caller is now responsible
  for maintaining the aggregated total. The caller provides a callback delegate when the ObservableCounter is created and the callback is invoked whenever tools need to observe
  the current total. For example, if a collection tool updates every three seconds, then the callback function will also be invoked every three seconds. Most tools will have both
  the total and rate of change in the total available. If only one can be shown, rate of change is recommended. If the callback returns 0 on the initial call, 7 when it is called

>   again after three seconds, and 19 when called after six seconds then the tool will report those values unchanged as the totals. For rate of change, the tool will show 7-0=7

Reviewer: ```suggestion
  again after three seconds, and 19 when called after six seconds, then the tool will report those values unchanged as the totals. For rate of change, the tool will show 7-0=7
```

- For timing things, Histogram is usually preferred. Often it's useful to understand the tail of these distributions (90th, 95th, 99th percentile) rather than averages or
  totals.

> - Other common cases, such as business metrics, physical sensors, cache hit rates, or sizes of caches, queues, and files are usually well suited for `UpDownCounter` or `ObservableUpDownCounter`.

Reviewer 2: Cache hit rates might fit `ObservableGauge` (probably the same for many physical sensors and business metrics).

Author: I agree that ObservableGauge would work for those scenarios, but I intentionally didn't advertise it as a 2nd recommended option to make the overall guidance simpler and more concise. If you think we should be talking about it more happy to discuss.

Reviewer 2: Maybe consider "Other common cases, such as size of caches, queues and files are usually ..."? (in this way folks who model physical sensors such as temperature / humidity / air pressure won't find it surprising/misleading)

Reviewer 2: Yes, delete both "business metrics" and "physical sensors" (because most of these won't fit for UpDownCounter/ObservableUpDownCounter), OR keep them and mention "ObservableGauge".

Author: Done : )

  Choose between them depending on which is easier to add to the existing code: either an API call for each increment and decrement operation or a callback that will read the current value from
  a variable the code maintains.

> [!NOTE]

> > If you are using an older version of .NET or the DiagnosticSource NuGet package that doesn't support `UpDownCounter` and `ObservableUpDownCounter` (before version 7), `ObservableGauge` is

Reviewer: ```suggestion
> If you're using an older version of .NET or a DiagnosticSource NuGet package that doesn't support `UpDownCounter` and `ObservableUpDownCounter` (before version 7), `ObservableGauge` is
```
